### ISSUE TITLE
Installing from source on an M1 Mac

### ISSUE THREAD

**Issue by milancermak** [solana-labs/solana#20783]
TL;DR:
All I had to do to build from source on an M1 Mac is adding this line to `[dependencies]` in `programs/bpf_loader/Cargo.toml`:
```
openssl = { version = "0.10", features = ["vendored"] }
```

#### Problem

There's a problem when developing for Solana on an M1 Mac as is evident by various issues on here (e.g. #19743 or #17406) or by others [on Stack Overflow](https://stackoverflow.com/search?q=solana+openssl). There's not really a good solution, the [commonly accepted one](https://dev.to/nickgarfield/how-to-install-solana-dev-tools-on-an-m1-mac-kfn) seems to be to use Rosetta and reinstall the whole toolchain. I didn't want to go down that route so I tried cloning the repo and compiling from scratch.

#### Proposed Solution

I already had `brew install openssl` and `brew install coreutils` (due to the `greadlink` dependency in `cargo_install_all.sh`). The only thing I needed to change was adding openssl as a dependency to Cargo.toml in `bpf_loader` as mentioned above.

I'm new to Solana and I don't know enough to send a PR, but I was hoping you could incorporate this to offer native M1 builds for the community.

**Comment by element0xE** [comment]
Hello guys , I compiled solana-test-validator from source, and now I can start it correctly without any openssl problems. But when I try to deploy locally, it remains stuck at :

⠤ [0/1] Finalizing transaction

and then

Error: Deploying program failed: unable to confirm transaction. This can happen in situations such as transaction expiration and insufficient fee-payer funds
I'm on M1 as weel , are you facing the same issue ?
Same issue [18033](https://github.com/solana-labs/solana/issues/18033)
